### PR TITLE
Updating tests to use moq.netcore

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @kichalla

File: test/Microsoft.Extensions.Logging.Testing.Tests/project.json [hunk 1]
```diff
@@ -1,23 +1,26 @@
 {
-    "version": "1.0.0-*",
```
Comment (kichalla) [inline test/Microsoft.Extensions.Logging.Testing.Tests/project.json]: Tab size 4 spaces...no?

**Comment by kichalla** [comment]
:shipit:

File: test/Microsoft.Extensions.Logging.Testing.Tests/project.json (hunk 1 above)
Comment (pranavkm) [inline test/Microsoft.Extensions.Logging.Testing.Tests/project.json]: We've been using 2 spaces in all other places. E.g. 
https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging/project.json
https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc/project.json

**Pull request merged**